### PR TITLE
Added istio files 

### PR DESCRIPTION
**Pull request by jjvilleg**
Adding Virtual Service and Gateway that uses an ingress controller. 
Adding default destination rule and working on authentication policy for when tls is enabled.

### CODE REVIEW

File: chart/auth/templates/istio_virtual_service.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  hosts:
+  - "*"
+  gateways:
+  - auth-gateway
+  http:
+  - match:
+    - port: 80
+    route:
+    - destination:
+        host: release-name-auth
```
Comment (dshimo) [inline chart/auth/templates/istio_virtual_service.yaml]: release-name is hardcoded

File: chart/auth/templates/istio_gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    tls:
+      httpsRedirect: {{ .Values.istio.gateway.TLS.httpsRedirect }}
```
Comment (dshimo) [inline chart/auth/templates/istio_gateway.yaml]: test redirect if actually works

File: chart/auth/templates/istio_gateway.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    hosts:
+    {{ toYaml .Values.istio.gateway.hosts }}
+{{- if .Values.istio.gateway.TLS.enabled }}
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    tls:
```
Comment (dshimo) [inline chart/auth/templates/istio_gateway.yaml]: don't worry about this for right now. 
we'll table mTLS

File: chart/auth/templates/authentication-microprofile.yaml [hunk 4]
```diff
@@ -3,6 +3,8 @@ kind: Service
 metadata:
   name: {{ .Release.Name }}-{{ .Values.service.name }}
   labels:
```
Comment (dshimo) [inline chart/auth/templates/authentication-microprofile.yaml]: this is where the _helper.tpl will be inserted.
see example in inventory

File: chart/auth/templates/_helpers.tpl [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{{- define "auth.fullname" -}}
```
Comment (dshimo) [inline chart/auth/templates/_helpers.tpl]: our structure makes it so we don't need fullname override
add labels template instead

File: chart/auth/templates/istio_gateway.yaml (hunk 2 above)
Comment (jjvilleg) [reply chart/auth/templates/istio_gateway.yaml]: redirect doesn't work. seems to be an istio bug but removing for now.

File: chart/auth/templates/istio_gateway.yaml (hunk 3 above)
Comment (jjvilleg) [reply chart/auth/templates/istio_gateway.yaml]: Sounds good. Removed for now.

**Pull request merged**